crates/bevy_pbr/src/render_graph/forward_pipeline/forward.frag

```diff
@@ -27,33 +31,143 @@ layout(set = 3, binding = 0) uniform StandardMaterial_albedo {
+    return F_Schlick(f0, f90, LoH);
+}
+
+vec3 isotropicLobe(vec3 f0, float roughness, const vec3 h, float NoV, float NoL, float NoH, float LoH) {
+
+    float D = D_GGX(roughness, NoH, h);
+    float V = V_SmithGGXCorrelated(roughness, NoV, NoL);
+    vec3  F = fresnel(f0, LoH);
+
+    return (D * V) * F;
+}
+
+float Fd_Burley(float roughness, float NoV, float NoL, float LoH) {
+    float f90 = 0.5 + 2.0 * roughness * LoH * LoH;
+    float lightScatter = F_Schlick(1.0, f90, NoL);
+    float viewScatter  = F_Schlick(1.0, f90, NoV);
+    return lightScatter * viewScatter * (1.0 / PI);
+}
+
+vec3 computeDiffuseColor(const vec3 baseColor, float metallic) {
+    return baseColor * (1.0 - metallic);
+}
+
+#define MIN_N_DOT_V 1e-4
+
+float clampNoV(float NoV) {
+    // Neubelt and Pettineo 2013, "Crafting a Next-gen Material Pipeline for The Order: 1886"
+    return max(NoV, MIN_N_DOT_V);
+}
+
+# endif
+
 void main() {
-    vec4 output_color = Albedo;
+    vec4 output_color = vec4(1.0, 0.0, 0.0, 1.0); //Albedo;
 # ifdef STANDARDMATERIAL_ALBEDO_TEXTURE
     output_color *= texture(
         sampler2D(StandardMaterial_albedo_texture, StandardMaterial_albedo_texture_sampler),
         v_Uv);
 # endif
 
 # ifdef STANDARDMATERIAL_SHADED
-    vec3 normal = normalize(v_Normal);
-    vec3 ambient = vec3(0.05, 0.05, 0.05);
+    float roughness = pbr.x;
+    float metallic = pbr.y;
+    vec3 N = normalize(v_Normal);
+    vec3 V = normalize(CameraPos.xyz - v_Position.xyz);
+
+    vec3 F0 = vec3(0.04); 
+    F0 = mix(F0, output_color.rgb, metallic);
+
+    vec3 diffuseColor = computeDiffuseColor(output_color.rgb, metallic);
+
     // accumulate color
-    vec3 color = ambient;
+    vec3 light_accum = vec3(0.0);
     for (int i=0; i<int(NumLights.x) && i<MAX_LIGHTS; ++i) {
         Light light = SceneLights[i];
-        // compute Lambertian diffuse term
-        vec3 light_dir = normalize(light.pos.xyz - v_Position);
-        float diffuse = max(0.0, dot(normal, light_dir));
-        // add light contribution
-        color += diffuse * light.color.xyz;
+
+        vec3 lightDir = light.pos.xyz - v_Position.xyz;
+        vec3 L = normalize(lightDir);
+
+        float rangeAttenuation = getDistanceAttenuation(lightDir, light.attenuation);
+        
+        vec3 H = normalize(L + V);
+
+        float NdotL = clampNoV(dot(N, L));
+        float NdotV = clampNoV(dot(N, V));
+        float NoL = saturate(NdotL);
+        float NoH = saturate(dot(N, H));
+        float LoH = saturate(dot(L, H));
+
+        vec3 specular = isotropicLobe(F0, roughness, H, NdotV, NoL, NoH, LoH);
+        vec3 diffuse = diffuseColor * Fd_Burley(roughness, NdotV, NoL, LoH);
+
+        light_accum += ((diffuse + specular) * light.color.xyz) * (light.color.w * NdotL);
     }
-    output_color.xyz *= color;
+
+    output_color.xyz = light_accum;
+
+    // Gamma correction.
+    output_color.xyz = output_color.xyz / (output_color.xyz + vec3(1.0));
```

Comment:
Technically this is not gamma correction (i.e. correcting for the output color space transfer function) but is actually an artistic tone mapping operator.